### ISSUE TITLE
[Safety] Implement advanced governor strategies

### ISSUE THREAD

**Issue by prof-schacht** [coairesearch/hierachical_safety_governor#16]
## Overview
Transform the simple rule-based governor into a sophisticated decision-making system that can handle complex multi-referee inputs and coordinate interventions.

## Goal
Create a governor that effectively aggregates referee signals, makes nuanced intervention decisions, and learns from past interventions.

## Tasks
- [ ] Define governor interface:
  ```python
  class BaseGovernor(ABC):
      @abstractmethod
      def process_alerts(self, alerts: List[Alert]) -> List[Intervention]:
          pass
      
      @abstractmethod
      def should_escalate_to_human(self, situation: Dict) -> bool:
          pass
  ```
- [ ] Implement governor strategies:
  - [ ] **ThresholdGovernor**: Simple rule-based decisions
    - Configurable alert thresholds
    - Priority-based processing
    - Cooldown periods
  - [ ] **ConsensusGovernor**: Referee voting mechanisms
    - Weighted voting by referee confidence
    - Quorum requirements
    - Tie-breaking rules
  - [ ] **AdaptiveGovernor**: Learning from outcomes
    - Track intervention effectiveness
    - Adjust thresholds based on results
    - Pattern recognition for repeated issues
  - [ ] **HierarchicalGovernor**: Multi-level oversight
    - Local governors for agent subgroups
    - Regional governors for environment sections
    - Global governor for system-wide issues
- [ ] Add intervention capabilities:
  ```python
  class Intervention:
      agent_id: Optional[str]  # None for system-wide
      action: InterventionType  # pause, reset, modify_reward, etc.
      duration: Optional[int]
      parameters: Dict
  ```
- [ ] Implement intervention types:
  - [ ] Agent pause/resume
  - [ ] Reward modification
  - [ ] Action override
  - [ ] Communication blocking
  - [ ] Environment reset
  - [ ] Policy constraints
- [ ] Add human escalation:
  - [ ] Severity scoring for situations
  - [ ] Human notification system
  - [ ] Decision queuing
  - [ ] Override mechanisms
- [ ] Create configuration system:
  ```yaml
  defenses:
    - id: adaptive_governor
      impl: defenses.adaptive_governor:AdaptiveGovernor
      params:
        learning_rate: 0.01
        intervention_types: [pause, modify_reward]
        escalation_threshold: 0.9
        human_webhook: "http://localhost:8080/alert"
  ```

## Decision Logic Examples
- Single high-confidence alert → Immediate intervention
- Multiple low-confidence alerts → Investigate pattern
- Repeated similar alerts → Escalate to human
- Conflicting referee opinions → Use consensus voting

## Acceptance Criteria
- Processes 1000+ alerts/second
- Intervention decisions in <10ms
- Clear audit trail for all decisions
- Human escalation works reliably
- Learns from intervention outcomes

## Testing
- Stress test with alert floods
- Test each intervention type
- Verify learning improvements
- Human escalation drills

**Priority: High** - Central coordination mechanism

## Dependencies
- Requires #13 (referee system)
